### ISSUE TITLE
[5.2] --queue= required on queue:listen

### ISSUE THREAD

**Issue by derokorian** [laravel/framework#11980]
Our listener quit working when updating to 5.2, and now requires specifying the --queue=name option to run any jobs. 

    php artisan queue:listen

This command does nothing but used to run all queues. However if we change to 

    php artisan queue:listen --queue=name_of_queue

Now runs jobs but only for this one queue, and we have multiple queues so this is pretty annoying. I don't see anything in the docs about requiring the name of a queue always, and its certainly not in artisan help queue:listen as being requried.

**Comment by rballman** [comment]
This is happening to me too.  Please help!

**Comment by GrahamCampbell** [comment]
> This command does nothing but used to run all queues.

That's not true. It never did. It's always just run jobs from the "default" queue.

**Comment by derokorian** [comment]
> That's not true. It never did. It's always just run jobs from the "default" queue.

We have always used OnQueue($queue_name) and never has $queue_name == 'default', but have never had to use the --queue argument before?

**Comment by derokorian** [comment]
Also note, that there is no default value in the help text as with the other options, so I wouldn't expect it to default to a specific queue if the option is not provided.

**Comment by GrahamCampbell** [comment]
> Our listener quit working when updating to 5.2, and now requires specifying the --queue=name option to run any jobs.

I can't replicate this.

**Comment by GrahamCampbell** [comment]
Can you show me why this doesn't work please?

**Comment by derokorian** [comment]
Sure, let me figure out how to recreate what we saw before (not even sure what version we were on). 

OTOH if you say it never did this, maybe we had a bug before that managed to put everything in the default queue, and we never noticed?

PS. My apologies for being short in my replies, it took quite a while to figure out to add the flag, and I was just annoyed. Didn't meant to take it out on your here...

**Comment by GrahamCampbell** [comment]
> My apologies for being short in my replies

Don't worry about it. I'm not easily offended, and tbh, I prefer short replies. Makes reading them easier. ;)

**Comment by derokorian** [comment]
Ok, I've reverted to when this was "working" and you are correct, we had a bug that called OnQueue with an empty string. I believe this caused it to end up in the default queue and hence it was working without the --queue argument. Closing as not an issue, but I have a question now - Is there a way to have it listen on all queues, without having to specify the list of all available queues? IE something like --queue=* instead of --queue=backups,reminders,etc..
